tsconfig.json

```diff
@@ -0,0 +1,56 @@
+{
+  "compilerOptions": {
+    "target": "es2017",
+    "allowJs": true,
+    "moduleResolution": "node",
+    "allowSyntheticDefaultImports": true,
+    "noImplicitAny": true,
+    "noImplicitThis": true,
+    "alwaysStrict": true,
+    "strictNullChecks": true,
+    "strictPropertyInitialization": true,
+    "noFallthroughCasesInSwitch": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noEmitOnError": true,
```

Comment:
This fails the build in case of any TS build error occurs for both development and CI. 
It may be not convenient for everyone, especially for people who are new with TS, or those who just want to quickly experiment and not to fight with types. 

So, if anyone knows a way to only enable such behavior on Travis, hints/suggestions are welcome!